toolkit/tools/imagegen/configuration/fileconfig.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//
+
+package configuration
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// DestinationFileConfigList is a list of destination files to copy.
+type FileConfigList []FileConfig
+
+// FileConfig specifies options for how a file is copied in the target OS.
+type FileConfig struct {
+	// The file path in the target OS that the file will be copied to.
+	Path string `json:"Path"`
+
+	// The file permissions to set on the file.
+	Permissions *FilePermissions `json:"Permissions"`
+}
+
+var (
+	DefaultFileConfig = FileConfig{
+		Path:        "",
+		Permissions: nil,
+	}
+)
+
+func (l *FileConfigList) IsValid() (err error) {
+	if len(*l) <= 0 {
+		return fmt.Errorf("list is empty")
+	}
+
+	return nil
+}
+
+func (l *FileConfigList) UnmarshalJSON(b []byte) error {
+	var err error
+
+	// Try to parse as a single value.
+	var fileConfig FileConfig
+	err = json.Unmarshal(b, &fileConfig)
+	if err == nil {
+		*l = FileConfigList{fileConfig}
```

Comment:
Does it make sense to also call `IsValid()` here? I'm thinking probably now with the current implementation but what if we add more checking to the validation function?

Reply:
Fixed.

Reply:
Isnt this validated as part of FileConfig UnmarshalJSON?

Reply:
@jiria This is validating the list type not the individual items in the list.